base_vat_optional_vies/tests/test_res_partner.py

```diff
@@ -43,3 +45,17 @@ def test_validate_simple_vat_vies(self):
             self.partner.vat = "MXGODE561231GR8"
             self.partner.country_id = 156
             self.assertEqual(self.partner.vies_passed, False)
+
+    def test_validate_vies_passed_false_when_vat_set_to_false(self):
+        with mock.patch(self.vatnumber_path) as mock_vatnumber:
+            mock_vatnumber.check_vies.return_value = True
+            self.partner.vat = "ESB87530432"
+            self.partner.country_id = 20
+            self.assertEqual(self.partner.vies_passed, True)
+            self.partner.vat = False
+            self.assertEqual(self.partner.vies_passed, False)
+
+    def test_validate_wrong_vat_shows_simple_message(self):
+        with self.assertRaisesRegex(ValidationError, "does not seem to be valid"):
+            self.partner.vat = "ES11111111A"
+            self.partner.country_id = 20
```

Comment:
This test is probably wrong. Or, at least, deceptive. The codecov warning is telling you this line doesn't get exercised in any test, and it's a test itself!

I initially thought it was a problem in codecov itself, but the real problem is that line 60 already raises the exception and thus line 61 is never executed (ref https://github.com/OCA/oca-addons-repo-template/issues/142#issuecomment-1151447046).

So you should either move line 61 outside of the `assertRaisesRegex` context, remove it, or put it before.

Keep this in mind. Whenever we need an `assertRaises`, it's better to limit it to a single line of code if possible.